ledger/src/blockstore.rs

```diff
@@ -1921,6 +1922,127 @@ impl Blockstore {
+                                        {
+                                            return Err(BlockstoreError::IO(IOError::new(
+                                                ErrorKind::Other,
+                                                format!(
+                                                    "Invalid start_after signature: {}",
+                                                    start_after
+                                                ),
+                                            )));
+                                        }
+                                    }
+                                    _ => {
+                                        // Should never happen...
+                                        return Err(BlockstoreError::IO(IOError::new(
+                                            ErrorKind::Other,
+                                            "Unexpected transaction message encoding".to_string(),
+                                        )));
+                                    }
+                                }
+                            }
+                            _ => {
+                                // Should never happen...
+                                return Err(BlockstoreError::IO(IOError::new(
+                                    ErrorKind::Other,
+                                    "Unexpected transaction encoding".to_string(),
+                                )));
+                            }
+                        }
+                        (slot, Some(start_after))
+                    }
+                }
+            }
+        };
+
+        // Fetch the list of signatures that affect the given address
+        let first_available_block = self.get_first_available_block()?;
+        let mut address_signatures = vec![];
+        loop {
+            if address_signatures.len() >= limit {
+                address_signatures.truncate(limit);
+                break;
+            }
+
+            let mut signatures = self.find_address_signatures(address, slot, slot)?;
+            if let Some(start_after) = start_after {
+                if let Some(index) = signatures
+                    .iter()
+                    .position(|(_slot, signature)| *signature == start_after)
+                {
+                    address_signatures.append(&mut signatures.split_off(index + 1));
+                    signatures.clear();
+                } else {
+                    // Should never happen...
+                    warn!(
+                        "Blockstore corruption?  Expected to find {} in slot {}",
+                        start_after, slot
+                    );
+                }
+            }
+            address_signatures.append(&mut signatures);
+            start_after = None;
+
+            if slot == first_available_block {
+                break;
+            }
+            slot -= 1;
```

Comment:
Can't iterate backwards over slots :-/

Reply:
Yeah. I saw that.  Didn't help much due to the intra-slot lexical ordering of signatures.   But if/when we rework address_signatures_cf to avoid this, `IteratorDirection::Reverse` would be perfect